File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02938/p02938_num5_parsed.py
L, R = map(int, input().split())

MOD = 10**9 + 7

count = 0

for x in range(L, R+1):
    for y in range(x, R+1):
        if y % x == (y ^ x):
            count += 1

print(count % MOD)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02938/p02938_num5_parsed.cpp
#include <iostream>
using namespace std;

const int MOD = 1000000007;

int main() {
    long long L, R;
    cin >> L >> R;

    long long count = 0;
    long long pow2 = 1;
    for (int i = 0; i < 63; i++) {
        long long mask = (1LL << (i + 1)) - 1;
        long long upper = (R & mask) + 1;
        long long lower = (L & mask) + 1;
        count += (((upper * (upper - 1) / 2) % MOD) - ((lower * (lower - 1) / 2) % MOD) + MOD) % MOD * pow2 % MOD;
        count %= MOD;
        pow2 = (pow2 * 2) % MOD;
    }

    cout << count << endl;

    return 0;
}